osu.Game/Screens/Edit/Compose/Components/Timeline/Timeline.cs

```diff
@@ -197,6 +189,17 @@ protected override void Update()
         {
             base.Update();
 
+            if (!trackWasLoaded && track.IsLoaded)
+            {
+                trackWasLoaded = true;
+
+                MaxZoom = getZoomLevelForVisibleMilliseconds(500);
+                MinZoom = getZoomLevelForVisibleMilliseconds(10000);
+                defaultTimelineZoom = getZoomLevelForVisibleMilliseconds(6000);
```

Comment:
Updating `MaxZoom`/`MinZoom` could invoke `OnZoomChange`, which would update the beatmap's `TimelineZoom` property to `Zoom / defaultTimelineZoom`, but since `defaultTimelineZoom` is still zero, that would write `Infinity` to the property.

Simplest fix would be to update `defaultTimelineZoom` first and foremost:
```suggestion
                defaultTimelineZoom = getZoomLevelForVisibleMilliseconds(6000);
                MaxZoom = getZoomLevelForVisibleMilliseconds(500);
                MinZoom = getZoomLevelForVisibleMilliseconds(10000);
```